### PR TITLE
Int extension: isOdd, isEven 

### PR DESCRIPTION
**Pull request by AndreasShortcut**
Made this for Nutrition AI, but need it in NMYN as well.

### CODE REVIEW

**Review by lemonandlime — APPROVED** [review]

File: Sources/ShortcutFoundation/Extensions/Int+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//  Created by Andreas Lif on 2022-01-12.
+//  Copyright © 2021 Shortcut Scandinavia Apps AB. All rights reserved.
+//
+
+import Foundation
+
+public extension Int {
+    
+    var isEven: Bool {
+        guard self != 0 else { return true }
```
Comment (gabrielPeart) [inline Sources/ShortcutFoundation/Extensions/Int+Extension.swift:14]: @AndreasShortcut 

Could we try:

return self.isMultiple(of: 2)

File: Sources/ShortcutFoundation/Extensions/Int+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        return self % 2 == 0
+    }
+    var isOdd: Bool {
+        guard self != 0 else { return false }
```
Comment (gabrielPeart) [inline Sources/ShortcutFoundation/Extensions/Int+Extension.swift:18]: @AndreasShortcut could we try:

return !self.isEven

File: Tests/ShortcutFoundationTests/IntTests.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//  Copyright © 2021 Shortcut Scandinavia Apps AB. All rights reserved.
+//
+
+import Foundation
+import XCTest
+@testable import ShortcutFoundation
+
+final class IntTests: XCTestCase {
+    
+    let odds = [1, 23, 405, 6897]
```
Comment (gabrielPeart) [inline Tests/ShortcutFoundationTests/IntTests.swift:15]: You can add the odds array to the odds unit tests

File: Tests/ShortcutFoundationTests/IntTests.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    let evens = [0, 2, 14, 376, 5908]
```
Comment (gabrielPeart) [inline Tests/ShortcutFoundationTests/IntTests.swift:16]: Same here

**Review by gabrielPeart — CHANGES_REQUESTED** [review]
Good job and fantastic initiative to improve our frameworks! I deeply love it ❤️

I also love the fact that we are improving the test coverage 💪

Thank you.

I have added some comments to make the code cleaner

File: Sources/ShortcutFoundation/Extensions/Int+Extension.swift (hunk 1 above)
Comment (jnutting) [inline Sources/ShortcutFoundation/Extensions/Int+Extension.swift:14]: I don't see why we need a `guard` in here at all. In general it's good to minimize branching where possible, and I don't think that we really gain anything by avoiding a `%` operation in the `0` case, when at the same time we are adding a branch to every case. I have the same concern about the `isOdd` method.

Comment (AndreasShortcut) [reply Sources/ShortcutFoundation/Extensions/Int+Extension.swift]: Yes, that's maybe easier to understand. Fixed in 5fcd885

File: Sources/ShortcutFoundation/Extensions/Int+Extension.swift (hunk 2 above)
Comment (AndreasShortcut) [reply Sources/ShortcutFoundation/Extensions/Int+Extension.swift]: I did `!(self.isMultiple(of: 2))` instead in 5fcd885. What do you think?

File: Sources/ShortcutFoundation/Extensions/Int+Extension.swift (hunk 1 above)
Comment (AndreasShortcut) [reply Sources/ShortcutFoundation/Extensions/Int+Extension.swift]: @jnutting Now that you mention it, I don't see the point either. I remember it not handling 0 when I made this, but it seems to work fine now, it so I guess I made a mistake somewhere. Oh well! I changed to @gabrielPeart's suggestion in 5fcd885

File: Tests/ShortcutFoundationTests/IntTests.swift (hunk 3 above)
Comment (AndreasShortcut) [reply Tests/ShortcutFoundationTests/IntTests.swift]: @gabrielPeart Thanks! That's how I originally did it, then I tested something and forgot to put it back ☺️ Fixed in 37f1d85

File: Tests/ShortcutFoundationTests/IntTests.swift (hunk 4 above)
Comment (AndreasShortcut) [reply Tests/ShortcutFoundationTests/IntTests.swift]: Fixed in 37f1d85

**Review by jnutting — APPROVED** [review]

File: Tests/ShortcutFoundationTests/IntTests.swift (hunk 3 above)
Comment (AndreasShortcut) [reply Tests/ShortcutFoundationTests/IntTests.swift]: Changed again, to your suggestion, in 5fcbeba

File: Sources/ShortcutFoundation/Extensions/Int+Extension.swift (hunk 2 above)
Comment (AndreasShortcut) [reply Sources/ShortcutFoundation/Extensions/Int+Extension.swift]: Changed again, to your suggestion, in 5fcbeba

**Pull request merged**